server/src/main/java/org/apache/iotdb/db/utils/MemUtils.java

```diff
@@ -88,7 +86,7 @@ public static long getRecordSize(BatchInsertPlan batchInsertPlan, int start, int
         case TEXT:
           memSize += (end - start) * 8L;
           for (int j = start; j < end; j++) {
-            memSize += ((Binary[]) batchInsertPlan.getColumns()[i])[j].getLength();
+            memSize += RamUsageEstimator.sizeOf(((Binary[]) batchInsertPlan.getColumns()[i])[j]);
```

Comment:
Hi, this may slow down the write performance of TEXT data heavily. It's better to do sampling or calculate the constant header of Binary and byte[].

Reply:
Agreed.

Reply:
I have modified the way to estimate the size of Binary. In new way, the efficiency is nearly same with the origin way which approved that it'll not affect the performance of the engine.